README.md

```diff
@@ -33,108 +34,242 @@ Menu
+
+See the [full example below](#example-code) for an illustration of how `setup()` and `teardown()` work, how they can be useful in a realistic situation, and how you can pass information from `setup()` to the `default` function and `teardown()`. You can also find more information and examples in the k6 docs [here](https://docs.k6.io/docs/test-life-cycle#section-setup-and-teardown-stages).
+
+
+### Metrics, tags and groups
+
+By default k6 measures and collects a lot of metrics about the things your scripts do - the duration of different script iterations, how much data was sent and received, how many HTTP requests were made, what was their duration, and even how long did the TLS handshake of a particual HTTPS request take. You see a summary of those built-in metrics in the output when you run the simplest possible k6 test, e.g. `k6 run github.com/loadimpact/k6/samples/http_get.js`. More information about the different built-in metrics collected by k6 (and how some of them can be accessed from inside of the scripts) can be seen in the docs [here](https://docs.k6.io/docs/result-metrics).
+
+k6 also allows the creation of user-defined `Counter`, `Gauge`, `Rate` and `Trend` metrics. They can be used to more precisely track and measure a custom subset of the things that k6 measures by default, or anything else the user wants, for example tracking non-timing information that is returned from the remote system. You can find more information about them [here](https://docs.k6.io/docs/result-metrics#section-custom-metrics) and a description of their APIs [here](https://docs.k6.io/docs/k6metrics).
+
+Every measurement metric in k6 comes with a set of key-value tags attached. Some of them are automatically added by k6 - for example a particular `http_req_duration` metric may have the `method=GET`, `status=200`, `url=https://loadimpact.com`, etc. system tags attached to it. Others can be added by users - globally for a test run via the `tags` [option](https://docs.k6.io/docs/options), or individually as a [parameter](https://docs.k6.io/docs/params-k6http) in a specific HTTP request, websocket connection, `userMetric.Add()` call, etc.
+
+These tags don't show in the simple summary at the end of a k6 test (unless you reference them in a [threshold](#checks-and-thresholds)), but they are invaluable for filtering and investigating k6 test results if you use any of the [outputs](#outputs) mentioned below. k6 also supports simple hierarchical groups for easier code and result organization. You can find more information about groups and system and user-defined tags [here](https://docs.k6.io/docs/tags-and-groups).
```

Comment:
By default k6 measures and collects a lot of metrics about the things your scripts do - the duration of different script iterations, how much data was sent and received, how many HTTP requests were made, what was their duration, and even how long did the TLS handshake of a particual HTTPS request take. You see a summary of those built-in metrics in the output when you run the simplest possible k6 test, e.g. `k6 run github.com/loadimpact/k6/samples/http_get.js`. For more information about the different built-in metrics collected by k6 (and how some of them can be accessed from inside of the scripts) refer to the docs [here](https://docs.k6.io/docs/result-metrics).

Reply:
Duplicate (earlier version?) of the previous comment: https://github.com/loadimpact/k6/pull/765#pullrequestreview-154780161